src/components/Modal/EditTaskModal.tsx

```diff
@@ -0,0 +1,111 @@
+import { axios } from '../../utils/axios';
+import { useRouter } from 'next/navigation';
+import { useState } from 'react';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { Endpoints } from '@/consts';
+import { createTaskSchema } from '@/schema/task';
+import { zodResolver } from '@hookform/resolvers/zod';
+import BaseForm from '../BaseForm';
+import ModalWrapper from './ModalWrapper';
+import FormInput from '../Input/FormInput';
+import { twMerge } from 'tailwind-merge';
+import z from 'zod';
+
+type Props = {
+  show: boolean;
+  onClose: () => void; // Add onClose prop
+  taskID: string;
+};
+
+type FormData = z.infer<typeof createTaskSchema>;
```

Comment:
Create a new type of schema for edit (will have fields which might not exist on creation and all are optional